### PR TITLE
Fix build for kernels >= 4.15.0

### PR DESCRIPTION
**Pull request by BlueGone**

### CODE REVIEW

**Comment by BlueGone** [comment]
The package linux-api-header, which provide macros `KERNEL_VERSION` and `LINUX_VERSION_CODE`, seem stuck in version 4.14.8 under most distribution repositories, including mine (Arch Linux), causing the compilation to fail on kernel >= 4.15, even after this change. I guess it is the reason why Travis failed.
After doing manually the upgrade for linux-api-headers to match my kernel version, it work well for me on Linux 4.15.1-2-ARCH

**Comment by aplund** [comment]
This seems wrong. It sounds like your building against the libc headers and not the kernel ones. The standard DKMS on arch builds against each of the installed kernel headers in /lib/modules/$KERNEL_VERSION. Perhaps check your build set up and see if it's using /usr/include.

**Comment by firsttris** [comment]
had no issues building this with 4.15.1-2-ARCH.

linux-api-headers where already installed on my system - somehow
╰─>$ pacman -Ss linux-api-headers                                                                                                                                                                           09:24:59
core/linux-api-headers 4.14.8-1 [Installiert]
    Kernel headers sanitized for use in userspace
community/aarch64-linux-gnu-linux-api-headers 4.15-1
    Kernel headers sanitized for use in userspace (aarch64-linux-gnu)

this change fixed evdi for kernel >= 4.15 on my system

**Comment by xblitz** [comment]
actually this should be `#if KERNEL_VERSION(4, 15, 0) <= LINUX_VERSION_CODE` as mentioned in #117

File: module/evdi_connector.c [hunk 1]
```diff
@@ -85,7 +85,7 @@ static struct drm_encoder *evdi_best_single_encoder(struct drm_connector
 	int enc_id = connector->encoder_ids[0];
 
 	return drm_encoder_find(connector->dev,
-#if KERNEL_VERSION(4, 16, 0) <= LINUX_VERSION_CODE
+#if KERNEL_VERSION(4, 15, 0) < LINUX_VERSION_CODE
```
Comment (xblitz) [inline module/evdi_connector.c]: it should stay `<=` though or else it will skip version 4.15.0 specifically (and work from 4.15.1)

**Comment by displaylink-dkurek** [comment]
Thanks all for looking into this. We have now version which compiles on 4.15 and 4.16 on devel branch :)
Closing.

**Pull request closed without merging**